src/core_plugins/kibana/public/management/sections/objects/components/objects_table/objects_table.js

```diff
@@ -196,11 +206,8 @@ export class ObjectsTable extends Component {
   };
 
   onQueryChange = ({ query }) => {
-    // TODO: investigate why this happens at EUI level
-    if (isSameQuery(query, this.state.activeQuery)) {
```

Comment:
I think there's a bug in EUI in which a reference is being reused within the `query.ast` instance, causing this comparison to incorrectly return `true`.